moduleframework/mtf_scheduler.py

```diff
@@ -229,21 +248,40 @@ def __init__(self, args, unknown):
         core.print_debug("additionalAvocadoArg = {0}".format(
             self.additionalAvocadoArg))
 
+    def check_tests(self):
+        output = subprocess.check_output([self.AVOCADO, "list", "-V", "--"] + self.tests)
+        assert "TEST TYPES SUMMARY" in output
+        badstates = ["NOT_A_TEST", "MISSING", "ACCESS_DENIED", "BROKEN_SYMLINK"]
+        badtests = []
+        # remove fist line, it is header, and remove last 11 lines, what is test summary
+        for testline in output.split("\n")[1:-11]:
+            splitted = testline.strip().split(" ", 1)
+            if splitted[0] in badstates:
+                badtests.append(splitted[1].strip())
+        if badtests:
+
+            core.print_info("", "ERROR: There are bad tests:", "-------------")
+            core.print_info(*badtests)
+            exit(19)
 
     def avocado_run(self):
+        self.check_tests()
         self.json_tmppath = tempfile.mktemp()
         avocado_args = ["--json", self.json_tmppath]
         if self.args.xunit:
             avocado_args += ["--xunit", self.args.xunit]
-        return self.avocado_general(avocado_default_args=avocado_args)
+        return self.avocado_general(action=self.args.action, avocado_default_args=avocado_args)
 
-    def avocado_general(self, avocado_default_args=[]):
-        # additional parameters
-        # self.additionalAvocadoArg: its from cmd line, whats unknown to this tool
-        avocadoAction = [self.AVOCADO, self.args.action] + avocado_default_args
+    def avocado_general(self, action, avocado_default_args=[]):
+        """
+
+        :param action: what avocado action to run: run, list, ...
+        :param avocado_default_args: list avocado additional argument
+        :return: return code of executed avocado command
+        """
         rc=0
         try:
-            subprocess.check_call(avocadoAction + self.additionalAvocadoArg + self.tests)
+            subprocess.check_call([self.AVOCADO, action] + avocado_default_args + self.additionalAvocadoArg + self.tests)
```

Comment:
Can you use os.path.join? I would rewrite it to more pythonic style.

Reply:
@phracek where to use ``os.path.join`` ?
I can rewrite it to some more python way, but I have not idea what is better?
``check_call`` uses ``dict`` as command, so I'm creating dict there with all command params